src/dopri.ts

```diff
@@ -0,0 +1,149 @@
+import * as dopri5 from "./dopri5/stepper";
+import * as interpolator from "./interpolator";
+import {RhsFn, Stepper} from "./types";
+import * as utils from "./utils";
+
+// needed for ES5 - will be ~= Number.EPSILON in ES6
+const DBL_EPSILON = Math.pow(2, -52); // = 2.220446049250313e-16
+const STEP_FACTOR_MIN = 1e-4;
+
+export function integrate(rhs: RhsFn, y: number[],
+                          t0: number, t1: number) {
+    const solver = new Dopri(rhs, y.length);
+    solver.initialise(t0, y);
+    return solver.run(t1);
+}
+
+function integrationError(message: string, t: number) {
+    return new Error(`Integration failure: ${message} at ${t}`);
+}
+
+export class Dopri {
+    public stepper: Stepper;
+    public t: number = 0.0;
+    public h: number = 0.0;
+
+    // state
+    public nSteps: number = 0;
+    public nStepsAccepted: number = 0;
+    public nStepsRejected: number = 0;
+
+    public stiffNStiff: number = 0;
+    public stiffNNonstiff: number = 0;
+    public lastError: number = 0;
+
+    // Stuff to tune
+    public maxSteps: number = 10000;
+    public atol: number = 1e-6;
+    public rtol: number = 1e-6;
+    public stiffCheck: number = 0;
```

Comment:
Also wondering if these should be private - not sure if they're public because you want to let people re-assign them, or just because they should be be visible outside the class. If the former, I'm imagining something like:
```
const solver = new Dopri(rhs, n);
solver.initialise(t0, y);
const result = solver.run(t1);

solver.atol = [[something else]]
solver.initialise(t0, y)
const newResult = solver.run(t1)
```
But there it kind of looks like they are more natural as constructor properties.
If they shouldn't be re-assigned but you want them to be visible outside the class you could use the pattern
```
 private _x: number = 0;
    public get x(): number {
        return this._x;
    }
```

Reply:
Also I guess in general it'd be nice to have an interface for this class, then it'd make all this a bit clearer. If there was an interface then you could leave everything public in the class if you needed them to be for testing say, but only include the intended entry points in the interface. Although seeing as your tests are in js at the moment I guess you don't need the public modifiers to reference them in tests anyway.

In terms of the compiled js, where the interface won't exist, the convention is to prefix properties intended as private with an underscore. So you could have, e.g. `_step()` then it'd be really clear that the end user doesn't need to interact with it directly.

Reply:
Thanks - this is very useful!